File: integrations/yasa/YASA-Engine-sembridge/test/callchain/java/TestCallchain.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.Statement;

public class TestCallchain {
    
    public void testSqlInjection(String userInput) {
        try {
            Connection conn = DriverManager.getConnection("jdbc:mysql://localhost:3306/test");
            Statement stmt = conn.createStatement();
            // This should be detected as a sink match
            String query = "SELECT * FROM users WHERE name = '" + userInput + "'";
            stmt.executeQuery(query);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
    
    public void testCommandInjection(String command) {
        try {
            // This should be detected as a sink match
            Runtime.getRuntime().exec(command);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
    
    public static void main(String[] args) {
        TestCallchain test = new TestCallchain();
        test.testSqlInjection("admin");
        test.testCommandInjection("ls -la");
    }
}

File: integrations/yasa/YASA-Engine-sembridge/test/callchain/python/test_callchain.py
import os
import sqlite3
import subprocess

def test_sql_injection(user_input):
    """Test SQL injection sink detection"""
    conn = sqlite3.connect('test.db')
    cursor = conn.cursor()

    # This should be detected as a sink match
    query = f"SELECT * FROM users WHERE name = '{user_input}'"
    cursor.execute(query)

    results = cursor.fetchall()
    conn.close()
    return results

def test_command_injection(command):
    """Test command injection sink detection"""
    # This should be detected as a sink match
    result = subprocess.call(command, shell=True)
    return result

def test_os_system(command):
    """Test os.system sink detection"""
    # This should be detected as a sink match
    os.system(command)

def test_eval_injection(code):
    """Test eval sink detection"""
    # This should be detected as a sink match
    eval(code)

if __name__ == '__main__':
    test_sql_injection('admin')
    test_command_injection('ls -la')
    test_os_system('pwd')
    test_eval_injection('print("test")')
